index.html

```diff
@@ -646,6 +659,106 @@ <h2 id="constraints">Unconstrained Display Surface Selection</h2>
     </section>
 
 
+    <section>
+      <h2 id="changing-display-surfaces">Changing Display Surfaces</h2>
+
+
+      <p>
+        The user agent MUST NOT change the [=display surface=]
+        associated with a [=capture-session=] unless the user has
+        explicitly indicated that they want this change to be
+        performed by interacting with user agent or operating system.
+
+        If the user indicates a change of [=display surface=], the
+        user agent MUST run the following steps:
+      </p>
+
+
+      <ol>
+        <li>
+          <p>If the [=capture-sessions=] does not have an associated
+          {{CaptureController}}, <var>controller</var>, abort these steps.</p>
+        </li>
+
+
+        <li>
+          <p>If <var>controller.{{CaptureController/[[DisplaySurfaceChangeCallback]]}}</var>
+              is null, abort these steps.</p>
+        </li>
+
+
+        <li>
+          <p>Stop delivering frames from all sources in the
+            [=capture-source-set=] associated with the [=capture-session=].
+          </p>
+        </li>
+
+
+         <li>
+           <p>Queue tasks to stop all tracks connected to sources in the
+             [=capture-source-set=] associated with the [=capture-session=].
+          </p>
+        </li>
```

Comment:
The order in the PR as it currently stands:
1. Stop delivering frames.
2. Queue tasks to end the old tracks.
3. Synchronously make changes to internal slots.
4. Synchronously invoke the callback.

Perhaps it should be:
1. Stop delivering frames.
2. Queue task to:
  a. Make changes to internal slots.
  b. Invoke the callback.
3. Queue tasks to end the old tracks.

Wdyt?

Reply:
I've grouped the synchronous steps into a queued task, and moved the ending of tracks to the end (+some other minor improvements)